### ISSUE TITLE
[macOS]: Missing privacy manifest

### ISSUE THREAD

**Issue by kjxbyz** [flutter/flutter#154915]
### Steps to reproduce

...

### Expected results

After uploading to the Mac App Store, you will not receive email notifications about issues with your submission.

### Actual results

```
ITMS-91061: Missing privacy manifest - Your app includes “Contents/Frameworks/FlutterMacOS.framework/Versions/A/FlutterMacOS”, which includes Flutter, an SDK that was identified in the documentation as a privacy-impacting third-party SDK. Starting November 12, 2024, if a new app includes a privacy-impacting SDK, or an app update adds a new privacy-impacting SDK, the SDK must include a privacy manifest file. Please contact the provider of the SDK that includes this file to get an updated SDK version with a privacy manifest. For more details about this policy, including a list of SDKs that are required to include signatures and manifests, visit: https://developer.apple.com/support/third-party-SDK-requirements.
```

### Code sample

<details open><summary>Code sample</summary>

```dart
Not provided
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.24.2, on macOS 14.6.1 23G93 darwin-x64, locale zh-Hans-CN)
    • Flutter version 3.24.2 on channel stable at /Users/ying/fvm/versions/3.24.2
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cf269e36d (7 天前), 2024-09-03 14:30:00 -0700
    • Engine revision a6bd3f1de1
    • Dart version 3.5.2
    • DevTools version 2.37.2

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/ying/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • ANDROID_SDK_ROOT = /Users/ying/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] IntelliJ IDEA Ultimate Edition (version 2024.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 81.1.3
    • Dart plugin version 242.21829.3

[✓] VS Code (version 1.93.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.96.0

[!] Proxy Configuration
    • HTTP_PROXY is set
    ! NO_PROXY is not set

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 14.6.1 23G93 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 128.0.6613.120
    ! Error: Browsing on the local area network for Kimmy's iPhone. Ensure the device is unlocked and attached with a cable or associated with the same local area network as
      this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by darshankawar** [comment]
I think https://github.com/flutter/flutter/issues/143381 would help to resolve this. 
I will keep it open for @stuartmorgan's input.

**Comment by kjxbyz** [comment]
I have asked Apple Support to find out if a privacy manifest is required for macOS, but have not received a response. I will update the status of this issue when I hear back.

**Comment by darshankawar** [comment]
For reference https://github.com/flutter/flutter/issues/143381#issuecomment-2340969960

**Comment by jmagman** [comment]
https://github.com/flutter/engine/pull/55078 is available in the [Flutter master channel](https://github.com/flutter/flutter/blob/master/docs/releases/Flutter-build-release-channels.md#how-to-change-channels). We would really appreciate it if someone could confirm this makes the warning go away.

**Comment by jmagman** [comment]
Stable hotfix request https://github.com/flutter/engine/pull/55366

**Comment by jmagman** [comment]
Closing, this is available on 3.24.4. 
https://github.com/flutter/flutter/blob/master/CHANGELOG.md#3244

Please continue to file new issues if you see App Store warnings/errors. Thank you!
